### PR TITLE
Make parafac() robust to complex tensors

### PR DESCRIPTION
**Pull request by maximeguillaud**
This makes PARAFAC correctly handle tensors with complex factors.

### CODE REVIEW

File: tensorly/cp_tensor.py [hunk 1]
```diff
@@ -598,7 +598,7 @@ def unfolding_dot_khatri_rao(tensor, cp_tensor, mode):
     _, rank = _validate_cp_tensor(cp_tensor)
     weights, factors = cp_tensor
     for r in range(rank):
-        component = multi_mode_dot(tensor, [f[:, r] for f in factors], skip=mode)
+        component = multi_mode_dot(tensor, [f[:, r].conj() for f in factors], skip=mode)
```
Comment (yngvem) [inline tensorly/cp_tensor.py:601]: This should probably be `tl.conj(f[:, r])`

**Comment by yngvem** [comment]
Nice addition :)

My only comment is that we should maybe add a test with a complex tensor before merging.

**Comment by JeanKossaifi** [comment]
Thanks @maximeguillaud! Looks ready to me too after making the changes mentioned by @yngvem

**Comment by JeanKossaifi** [comment]
Looks good to me aside from the test - if anyone has time to look at this would be great, it's just missing a test. Otherwise I'll look at it in a couple days, would be good to merge if it fixes a bug in complex support. @caglayantuna @yngvem

**Pull request merged**